apis/config/v1alpha1/feature_types.go

```diff
@@ -40,6 +40,7 @@ type FeatureStatus struct{}
 // +kubebuilder:printcolumn:name="Description",type=string,JSONPath=.spec.description
 // +kubebuilder:printcolumn:name="Immutable",type=boolean,JSONPath=.spec.immutable
 // +kubebuilder:printcolumn:name="Maturity",type=string,JSONPath=.spec.maturity
+// +kubebuilder:deprecatedversion:warning="Feature API in config.tanzu.vmware.com is deprecated. Use Feature API from core.tanzu.vmware.com instead"
```

Comment:
Nit: sometimes in kubebuilder markers there is a space after the `//` comment tags and other times there aren't. I think we should just choose a format and stick with it. I think it's more common not to have that space, so I say remove these spaces, but this is a nit, also, so your choice.